File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
# def solution(phone_book):
#     answer = True
#     #어떤 번호가 다른 번호의 접두어인 경우가 있으면 false를 그렇지 않으면 true를 return
#     phone_book.sort()
#     for i in range(len(phone_book)-1)  :
#         if  phone_book[i]== phone_book[i+1][0:(len(phone_book[i]))]:
#             answer= False
#             break
#         else:
#             continue
    
#     return answer

def solution(phone_book):
    answer=''
    #문자열 기준으로 정렬 + 다음거랑 비교
    phone_book.sort()

    for i in range(len(phone_book)-1):
        length= len(phone_book[i])
        if len(phone_book[i])<=len(phone_book[i+1]):       
            if phone_book[i+1][0:length]==phone_book[i]:
                return False
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
//번호 array phone_book에서 어떤 번호가 다른 번호의 접두어인지 확인 접두어면 f, 아니면 t
function solution(phone_book) {
    let answer = true;
    phone_book.sort()
    for(let i=0; i<phone_book.length-1; i++){
        if (phone_book[i]==phone_book[i+1].slice(0,phone_book[i].length)){
            answer=false
            break
        }
    }
    return answer;
}